pkg/instrumentation/database.go

```diff
@@ -73,16 +78,15 @@ func (c *callbacks) before(db *gorm.DB, operationName string, serviceName string
 		operationName = strings.Split(db.Statement.SQL.String(), " ")[0]
 	}
 	sp := ddtrace.StartSpan(operationName, spanOpts...)
-	db.Set(SpanGormKey, sp)
+	db.Statement.Context = context.WithValue(db.Statement.Context, spanGormKey, sp)
 }
 
 func (c *callbacks) after(db *gorm.DB) {
-	val, ok := db.Get(SpanGormKey)
+	sp, ok := db.Statement.Context.Value(spanGormKey).(ddtrace.Span)
```

Comment:
In the [dd-trace-go](https://github.com/DataDog/dd-trace-go/blob/48141787da9e038992e67c66768b722f1c7713c8/contrib/gorm.io/gorm.v1/gorm.go#L103) there is a safety check. Shouldn't we place the same check here?

Reply:
Although it's covered by tests but I added it to make sure we won't get panic :+1: